File: Project Templates/TenPearls/iOS Starter App.xctemplate/objc/Components/Label.m
//___FILEHEADER___

#import "Label.h"
#import "Constant.h"

@implementation Label

-(void)awakeFromNib{
    [super awakeFromNib];
    
    NSString *font = FONT_MATRIX[self.isBold][self.isCondensed];
    
    self.font = [UIFont fontWithName:font size:self.font.pointSize];
}

@end

File: Project Templates/TenPearls/iOS Starter App.xctemplate/swift/Controls/Label.swift
// ___FILEHEADER___.

import UIKit

class Label: UILabel {
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        setFont()
    }
    
}

extension UILabel {
    
    func setFont() {
        if let face = font.fontDescriptor.object(forKey: .face) as? String {
            font = Fonts.font(face: face, fontSize: font.pointSize)
        }
        
    }
    
}
